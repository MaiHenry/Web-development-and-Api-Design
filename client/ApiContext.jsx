import React from "react";
import { postJSON } from "./components/apiRequests/postJSON";
import { fetchJSON } from "./components/apiRequests/fetchJSON";
import { putJSON } from "./components/apiRequests/putJSON";

// ApiContext for chat related API
export const ApiContext = React.createContext({
  fetchChatRooms: async () => {
    return await fetchJSON("/api/chatroom");
  },
  createRoom: async (roomData) => {
    return await postJSON("/api/chatroom/add", roomData);
  },
  fetchChatRoomById: async (roomId) => {
    return await fetchJSON(`/api/chatroom/${roomId}`);
  },
  updateChatRoom: async (roomId, roomData) => {
    return await putJSON(`/api/chatroom/${roomId}`, roomData);
  },
});