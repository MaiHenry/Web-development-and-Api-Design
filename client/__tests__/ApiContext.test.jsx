import React, { useContext } from "react";
import renderer from "react-test-renderer";
import { ApiContext } from "../ApiContext";

const TestingComponent = () => {
  const {
    fetchChatRooms,
    createRoom,
    fetchChatRoomById,
    updateChatRoom,
    deleteRoom,
    fetchMessages,
    postMessage,
  } = useContext(ApiContext);

  return (
    <div>
      <button onClick={fetchChatRooms}>Fetch Rooms</button>
      <button onClick={() => createRoom({ name: "BEST Room" })}>
        Create Room
      </button>
      <button onClick={() => fetchChatRoomById("HAHAHA")}>
        Fetch Room By ID
      </button>
      <button
        onClick={() => updateChatRoom("HAHAHA", { name: "Updated Room" })}
      >
        Update Room
      </button>
      <button onClick={() => deleteRoom("HAHAHA")}>Delete Room</button>
      <button onClick={() => fetchMessages("HAHAHA")}>Fetch Messages</button>
      <button onClick={() => postMessage("HAHAHA", { text: "New Message" })}>
        Post Message
      </button>
    </div>
  );
};

describe("ApiContext", () => {
  it("TestingComponent W/ context", () => {
    const mockFunctions = {
      fetchChatRooms: jest.fn(),
      createRoom: jest.fn(),
      fetchChatRoomById: jest.fn(),
      updateChatRoom: jest.fn(),
      deleteRoom: jest.fn(),
      fetchMessages: jest.fn(),
      postMessage: jest.fn(),
    };

    const component = renderer.create(
      <ApiContext.Provider value={mockFunctions}>
        <TestingComponent />
      </ApiContext.Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
