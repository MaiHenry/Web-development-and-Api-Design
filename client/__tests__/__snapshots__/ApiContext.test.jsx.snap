// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiContext TestingComponent W/ context 1`] = `
<div>
  <button
    onClick={[MockFunction]}
  >
    Fetch Rooms
  </button>
  <button
    onClick={[Function]}
  >
    Create Room
  </button>
  <button
    onClick={[Function]}
  >
    Fetch Room By ID
  </button>
  <button
    onClick={[Function]}
  >
    Update Room
  </button>
  <button
    onClick={[Function]}
  >
    Delete Room
  </button>
  <button
    onClick={[Function]}
  >
    Fetch Messages
  </button>
  <button
    onClick={[Function]}
  >
    Post Message
  </button>
</div>
`;
